Add "state x days ago" query

Marquo1 opened the pull request:
I think a sensor showing a state 1 day ago is potentially a common use case for sql integration and should be in the docs to promote its use.

I did not write the query, it comes from: https://community.home-assistant.io/t/sensor-state-24-h-ago/524994/13?u=marquo 
Since I lack knowledge and don't want to ruin it, I copied it without changing the formatting. Please adjust if needed.

## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

Marquo1 committed:
Add a state x days ago query

This is my second attempt on proposing this change, my apologies if the first time went through.

I think a sensor showing a state 1 day ago is potentially a common use case and should be in the docs to promote its use.

I did not write the query, it comes from: https://community.home-assistant.io/t/sensor-state-24-h-ago/524994/13?u=marquo
Since I lack knowledge and don't want to ruin it, I copied it without changing the formatting.

frenck commented on source/_integrations/sql.markdown:
```diff
@@ -185,6 +185,21 @@ WHERE
 
+### State of an entity x time ago
+
+If you want to extract the state of entity from a day, hour or minute ago, the query is:
```
```suggestion
If you want to extract the state of an entity from a day, hour, or minute ago, the query is:

```

frenck commented on source/_integrations/sql.markdown:
````diff
@@ -185,6 +185,21 @@ WHERE
+
+If you want to extract the state of entity from a day, hour or minute ago, the query is:
+```sql
+SELECT state 
+FROM states 
+INNER JOIN states_meta ON states.metadata_id = states_meta.metadata_id
+WHERE states_meta.entity_id = 'sensor.temperature_in' 
+  AND last_updated_ts <= strftime('%s', 'now', '-1 day')
+ORDER BY last_updated_ts DESC LIMIT 1;
````
Can we format the SQL query just like all other queries on this page? It will help with readability.

Marquo1 replied:
I lack knowledge about query structure to make correct indentations myself

Marquo1 replied:
Done

frenck commented on source/_integrations/sql.markdown:
````diff
@@ -185,6 +185,21 @@ WHERE
+  AND last_updated_ts <= strftime('%s', 'now', '-1 day')
+ORDER BY last_updated_ts DESC LIMIT 1;
+```
+Replace  `-1 day` with the target offset, e.g. `-1 hour`.
````
```suggestion

Replace `-1 day` with the target offset, for example, `-1 hour`.
```

Marquo1 committed:
Update source/_integrations/sql.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Marquo1 committed:
Update source/_integrations/sql.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Marquo1 committed:
Added required input for column 

Added column input for improved clarity and uniformity with the rest of the document.

Marquo1 committed:
Update sql.markdown

Changed formatting of the query

Marquo1 committed:
Merge branch 'current' into patch-2

Marquo1 committed:
Update sql.markdown

Marquo1 committed:
Merge branch 'current' into patch-2

gjohansson-ST commented on source/_integrations/sql.markdown:
````diff
@@ -185,6 +185,30 @@ WHERE
+
+```sql
+SELECT 
+  state 
````
```suggestion
  states.state
```

gjohansson-ST commented on source/_integrations/sql.markdown:
```diff
@@ -185,6 +185,30 @@ WHERE
+  AND last_updated_ts <= strftime('%s', 'now', '-1 day')
+ORDER BY 
+  last_updated_ts DESC 
+LIMIT 1;
```
```suggestion
LIMIT
  1;
```

Marquo1 committed:
Update source/_integrations/sql.markdown

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

Marquo1 committed:
Update source/_integrations/sql.markdown

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

Marquo1 committed:
Merge branch 'current' into patch-2

Marquo1 committed:
Merge branch 'current' into patch-2

frenck reviewed (approved):
Thanks, @Marquo1 👍 

../Frenck
